bin/warehouse/generate_warehouse.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+"""
+Creates Warehouse for all PD-Types found in archived backups.
+
+Arguments:
+fname - directory of archived backups
+operation - '-d' to compare last 2 backups (default), '-a' to compare all backups.
+"""
+import tarfile
+import sys
+import os
+import subprocess
+import shutil
+from datetime import datetime
+
+fname = sys.argv[1]
+
+if len(sys.argv) == 3:
+    operation = sys.argv[2]
+    if operation != '-a' and operation != '-d':
+        sys.exit("Error: invalid operation value (sys.argv[2]), -d or -a expected")
+else:
+    operation = "-d"
+
+tar_array = sorted(os.listdir(fname))
+if operation == "-d":
+    tar_array = tar_array[-2:]
+print(tar_array)
+prev = ''
+curr = ''
+
+def get_base(tfile):
+    base = os.path.basename(tfile)
+    pd_name = os.path.splitext(os.path.splitext(base)[0])[0]
+    return pd_name
+
+def extract(tfile, dest):
+    fpath = './' + dest
+    tar = tarfile.open(fname + tfile)
+    tar.extractall(path=fpath)

```

Comment:
you can operate on tarfiles directly with the tarfile module, there's no need to extract to a temporary directory.

Reply:
Looking at the tarfile module doc, eventually the files would need to be extracted individually to be worked on anyways, whether it's here or later in the migrate_all.py file. Wouldn't it be better to extract_all at this step? It would be simpler for then running the python subprocess command too at line 73.